test/integration/ovs/ofctrl_test.go

```diff
@@ -268,6 +270,85 @@ func TestTransactions(t *testing.T) {
 	}
 }
 
+func TestBundleErrorWhenOVSRestart(t *testing.T) {
+	br := "br06"
+	err := PrepareOVSBridge(br)
+	require.Nil(t, err, fmt.Sprintf("Failed to prepare OVS bridge: %v", err))
+	defer func() {
+		err = DeleteOVSBridge(br)
+		require.Nil(t, err, fmt.Sprintf("error while deleting OVS bridge: %v", err))
+	}()
+
+	bridge := binding.NewOFBridge(br)
+	table = bridge.CreateTable(2, 3, binding.TableMissActionNext)
+
+	err = bridge.Connect(maxRetry, make(chan struct{}))
+	require.Nil(t, err, "Failed to start OFService")
+	defer bridge.Disconnect()
+
+	// Ensure OVS is connected before sending bundle messages.
+	select {
+	case <-time.Tick(1 * time.Second):
+		if bridge.IsConnected() {
+			break
+		}
+	}
+
+	// Restart OVS in another goroutine.
+	go func() {
+		select {
+		case <-time.After(100 * time.Millisecond):
+			DeleteOVSBridge(br)
+			PrepareOVSBridge(br)
+		}
+	}()
+
+	var failCount, successCount int
+	loop := 10000
+	var wg sync.WaitGroup
+	wg.Add(loop)
+	i := 0
+	for i < loop {
+		// Sending Bundle message in parallel.
+		go func() {
+			// Sending OpenFlow messages when OVS is disconnected is not in this case's scope.
+			if !bridge.IsConnected() {
+				return
+			}
+			ch := make(chan struct{})
+			go func() {
+				flows := []binding.Flow{table.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+					Cookie(getCookieID()).
+					MatchInPort(uint32(count + 1)).
+					Action().ResubmitToTable(table.GetNext()).
+					Done()}
+				err = bridge.AddFlowsInBundle(flows, nil, nil)
+				if err != nil {
+					// Check if it is a timeout error
+					require.True(t, strings.Contains(err.Error(), "timeout"))
+				}
+				ch <- struct{}{}
+			}()
+
+			select {
+			// The timeout for both message sending and waiting for the reply are 10s. Test should fail if the bundle
```

Comment:
Isn't 10 seconds too long? It means if ovs is restarted and a bundle happens to be installed, ReplayFlows will have to wait for 10 seconds during which Pods will communicate freely locally because of the normal flow.
Or can we use a context to tell bundle functions that ovs is disconnected so they can terminate immediately, then 10 seconds will not be too long as it's just an insurance for unknown cases.

Reply:
I didn't find the default timeout value of ovs-ofctl, 10s is from ovs-vsctl command.

Reply:
> > @tnqn I am thinking it is dangrous to restart OVS for it is possible to affect other cases, e.g., when OVS is restarted, there are some other tests are installing OpenFlow messages. So I adjust the test case to remove the bridge and then add a new one with same name. Then the UDS path for the OF Switch should be the same. But since the datapath-id has changed, there exist disconnection and reconnection events. And this should be able to simulate OVS restart event. What do you think?
> 
> I don't know integration tests run in parallel, I see there are already disruptive tests, did you see they run in parallel?
I checked the failure messages, it looks some other cases are affected by OVS restart(not able to connect OVS bridge).The failed case is using a different bridge name which is not used in the new case.

Reply:
It sounds good to use a context to cancel messages when OVS is disconnected. I will update ofnet to leverage context for both sending messages and waiting for reply.